File: backend/app/models/challan.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Float,
    ForeignKey,
    Date,
)

from sqlalchemy.orm import relationship

from app.db.database import Base


class Challan(Base):

    __tablename__ = "challans"

    id = Column(Integer, primary_key=True, index=True)

    challan_number = Column(
        String,
        unique=True,
        nullable=False
    )

    vendor_id = Column(
        Integer,
        ForeignKey("vendors.id")
    )

    supplier_id = Column(
        Integer,
        ForeignKey("suppliers.id")
    )

    quantity = Column(Float, nullable=False)

    unit = Column(String, nullable=False)

    transport_type = Column(String)

    lr_number = Column(String)

    program = Column(String)

    rate = Column(Float)

    sent_date = Column(Date)

    status = Column(String)

    vendor = relationship(
        "Vendor",
        back_populates="challans"
    )

    supplier = relationship(
        "Supplier",
        back_populates="challans"
    )

    returns = relationship(
        "ChallanReturn",
        back_populates="challan"
    )

File: frontend/src/types/challan.ts
export interface Challan {

  id?: number;

  challan_number?: string;

  vendor_id: number;

  supplier_id: number;

  vendor_name?: string;

  supplier_name?: string;

  quantity: number;

  returned_quantity?: number;

  pending_quantity?: number;

  unit: string;

  transport_type?: string;

  lr_number?: string;

  program?: string;

  rate?: number;

  sent_date: string;

  status?: string;
}